ui/components/MesherySettings.js

```diff
@@ -129,6 +139,64 @@ class MesherySettings extends React.Component {
     return null;
   }
 
+  componentDidMount() {
+    this.fetchPromGrafanaScanData()
+  }
+
+  fetchPromGrafanaScanData = () => {
+    const self = this;
+    self.props.updateProgress({ showProgress: true });
+    dataFetch(
+      '/api/promGrafana/scan', 
+      {
+        credentials: "same-origin",
+        method: "GET",
+        credentials: "include",
+      },
+      (result) => {
+        self.props.updateProgress({ showProgress: false });
+        if (Array.isArray(result.prometheus)) {
+          result.prometheus.forEach(prom => {
+            if (Array.isArray(prom.status?.loadBalancer?.ingress)) {
+              const urls = prom.status.loadBalancer.ingress.map((ip, i) => {
+                return `http://${ip.ip}:${prom.spec?.ports[i]?.port}`
+              })
+
+              self.setState(state => ({ scannedPrometheus: [...state.scannedPrometheus, ...urls] }))
+            }
+          })
+        }
+
+        if (Array.isArray(result.grafana)) {
+          result.grafana.forEach(graf => {
+            if (Array.isArray(graf.status?.loadBalancer?.ingress)) {
+              const urls = graf.status.loadBalancer.ingress.map((ip, i) => {
+                return `http://${ip.ip}:${graf.spec?.ports[i]?.port}`
```

Comment:
A potential issue with secure endpoints (https)

Reply:
@leecalcote, Actually I couldn't find a reliable way to determine "http" and "https". Should I assign "https" if the port is 443 and http for the rest? But I think that other ports can also be used other than 443 for https, not sure about that though.

Reply:
@utkarsh-pro you're right. There really isn't a definitive way to identify this protocol… Or at least not an easy one. Yes, the logic that you highlighted regarding HTTPS for 443 and HTTP for everything else will work for the vast majority of users. Refining this overtime is something that we can do with user feedback.